libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2680,6 +2654,14 @@ VariableDeclaration const* SMTEncoder::identifierToVariable(Expression const& _e
 			solAssert(m_context.knownVariable(*varDecl), "");
 			return varDecl;
 		}
+	// But we are interested in "contract.var", because that is the same as just "var".
```

Comment:
Isn't this case already handled by `memberAccess`?

Reply:
As discussed, `visit` for `memberAccess` handles the creation of SMT expression for such solidity Expression, but in here I need to get to the state variable's `VariableDeclaration`.